lib/active_storage/engine.rb

```diff
@@ -48,6 +48,11 @@ class Engine < Rails::Engine # :nodoc:
             raise "YAML syntax error occurred while parsing #{config_file}. " \
                   "Please note that YAML must be consistently indented using spaces. Tabs are not allowed. " \
                   "Error: #{e.message}"
+          rescue NoMethodError => e
+            abort "ERB error while parsing #{config_file}. " \
+                  "If you're using AWS for storage, add its keys into your secrets file. " \
+                  "If not, remove Amazon from storage_services.yml. " \
+                  "Error: #{e.message}"
```

Comment:
@georgeclaghorn As an alternative, we can ignore the error here and let the `Aws::S3::Resource` fail on first upload. Or even check manually in S3Service's constructor. All are lousy solutions, though.

The proper solution would be for `Rails.application.secrets` to raise something useful instead of `undefined method`.